velox/exec/WindowFunction.cpp

```diff
@@ -76,15 +76,22 @@ std::unique_ptr<WindowFunction> WindowFunction::create(
 void WindowFunction::setNullEmptyFramesResults(
     const SelectivityVector& validRows,
     vector_size_t resultOffset,
-    const VectorPtr& result) {
+    const VectorPtr& result,
+    const std::optional<VectorPtr> defaultValue) {
   if (validRows.isAllSelected()) {
     return;
   }
 
   invalidRows_.resizeFill(validRows.size(), true);
   invalidRows_.deselect(validRows);
-  invalidRows_.applyToSelected(
-      [&](auto i) { result->setNull(resultOffset + i, true); });
+  if (defaultValue != nullptr) {
+    invalidRows_.applyToSelected([&](auto i) {
+      result->copy(defaultValue.value().get(), resultOffset + i, 0, 1);
```

Comment:
Isn't defaultValue.value() API only available for a ConstantVector ? This code seems to have some underlying assumptions about it.

Also, isn't value only for scalar types ? How are you handling complex types in this code ?

Reply:
Thanks for pointing it out, the optional parameter and use of `value()` is removed.